gesifred/rs-nodejs-b
# all solutions

**NOTE**:
https://github.com/gesifred/rs-nodejs-b/pull/1#issuecomment-1599116730
**Basic Scope**
 -     File system (src/fs)

        +6 create.js implemented properly
        +10 copy.js implemented properly
        +10 rename.js implemented properly
        +6 delete.js implemented properly
        +6 list.js implemented properly
        +6 read.js implemented properly

-     Command line interface(src/cli)

        +6 env.js implemented properly
        +6 args.js implemented properly

-     Modules(src/modules)

        +20 cjsToEsm.cjs refactored properly

-     Hash (src/hash)

        +10 calcHash.js implemented properly

-     Streams (src/streams)

        +10 read.js implemented properly
        +10 write.js implemented properly
        +10 transform.js implemented properly

-     Zlib (src/zip)

        +10 compress.js implemented properly
        +10 decompress.js implemented properly

- Advanced Scope

-     - Worker Threads (src/wt)

        +10 worker.js implemented properly
        +30 main.js implemented properly

-     - Child Processes (src/cp)

        +10 spawns child process
        +10 child process stdin receives input from master process stdin
        +10 child process stdout sends data to master process stdout

# Module "Node.js basics"

## Technical requirements

- Use 18 LTS version of Node.js

# How to check student's project

1. Clone repository with the project to check with `git clone [HTTPS]` or download it.
   (where HTTPS is the GitHub URL of the project to be used to clone. To get it:
   a) go to GitHub repository that you want to clone;
   b) click _Code_ and copy given URL.)

2. Open following files:

3. Open project in Visual Studio Code or any other code editor.

4. Switch from main branch to branch with solution, if you haven't done it yet.

5. Start to check subtasks according to the assignment description.
   For check simplification students expected to have npm-scripts in `package.json` how to run every subtask.
   If not, follow below instruction:

All tasks are located inside `src` folder. Open terminal in **src** folder of downloaded project.
For example 1st subtask **create.js** located in _fs_ folder:

- Run `node fs/create.js` in your terminal opened in _src_ folder.
  or go into fs/ subfolder of the task and run `node create.js`.

## Steps to check subtask

- According to this subtask after running this function there should be file named `fresh.txt` created in _files_ folder ( At the beggining there was no such file, if found - delete it).
- Open this file `fresh.txt` and you have to see following text created by the function after you run it:
  _I am fresh and young_.
- Try to run `node fs/create.js` again and the error _FS operation failed_ must be thrown.
- If all above works as expected then add _+6_ points to the total score as written in `Score calculation`.
- If function works partly deduct points from the maximum allowed for this task.
  (For example function creates `fresh.txt` file, but on a second run it doesn't throw an error, half of task is executed only. Add only _+3_ points for this subtask).

- If function is not implemented at all mark it as _0_.

- Continue to check all other subtasks in a same way. Add points for each to grand total.

5.  Once you finish to check all subtasks then check for a forfeit if any:

- `package.json` should not have any external tools/libraries as dependencies.
  **-95% of total task score** Any external tools/libraries are used
- Check that there are no commits made after deadline.
  **-30% of total task score** Commits after deadline (except commits that affect only Readme.md, .gitignore, etc.).

6. Submit total score with details check description to cross-check section of this task at [App RS School](https://app.rs.school/) before deadline.
